src/psyclone/tests/utilities_test.py

```diff
@@ -49,99 +56,237 @@
 '''
 
 
-def test_compiler_works(tmpdir):
-    ''' Check that the specified compiler works for a hello-world
-    example '''
+def test_enable_disable_compilation(monkeypatch):
+    '''Test the behaviour when disabling compilation ... even if
+    compilation is enabled.
+    '''
+    monkeypatch.setattr(Compile, "TEST_COMPILE", False)
+    # Test that compile_file will do nothing if compilation is disabled:
+    _compile = Compile()
+    _compile.compile_file("nothing-to-compile")
+    _compile.code_compiles(None)
+    _compile.string_compiles("")
     Compile.skip_if_compilation_disabled()
```

Comment:
What does this actually test? Should we have something after this statement to check that the skip works?

Reply:
OK, I found a nice way (patching pytest.skip) to test that it really does the right thing. Thanks for this question!

Reply:
Aha. Thanks for extending the comments too. I must have been a bit slow last time around :-)

Reply:
No, you weren't. I first wrote a wrong comment when I updated it, then had to dig into the code again - these comments were clearly missing.